api/base/views.py

```diff
@@ -437,7 +470,36 @@ def root(request, format=None, **kwargs):
     if utils.has_admin_scope(request):
         return_val['meta']['admin'] = True
 
-    return Response(return_val)
+    resp = Response(return_val)
+
+    # This is used exclusively for our partnership with Sloan and can be deleted after their study is complete.
```

Comment:
We can't really mess around with it too much because 1) we need to instatiate the response before setting cookies (bad design choice IMO) and 2) even just creating a helper function here will take the flask request out of context and break, gerrr Flask.

Reply:
i thought flask request context was per-thread, specifically so multiple functions can access the same request without worrying about it?

Reply:
[It's this function that throws the errors](https://github.com/CenterForOpenScience/osf.io/blob/develop/api/waffle/utils.py#L12-L22) I swapped it out for the model level one that seems to work.

Reply:
oof, i see. the model-level one does seem much cleaner 👍